docs/NamespaceIsolation.md

```diff
@@ -0,0 +1,58 @@
+# Namespace isolation for BPF objects
+This document describes how eBPF for Windows implements namespace isolation.
+
+## Overview
+BPF objects are by default system wide entities that can be opened via several libbpf APIs (meaning the caller receives
+a fd to the object). While this is useful for the general case, there are some scenarios where BPF applications need to
+be able to isolate themselves from other BPF applications to ensure that their state remains consistent.
+
+To achieve this a BPF application can switch to a designated namespace. Namespaces are identified by a GUID, with the
+default namespace having the zero GUID (all fields in the GUID are zero). All libbpf APIs that return fds are relative
+to the namespace that the application has specified.
+
+## Design details
+Each BPF object will have a namespace identifier associated with it. Each new object created is marked with the current
+namespace identifier. All calls to the execution context to open objects are relative to the current namespace.
```

Comment:
Will the app choose / generate this namespace GUID? If yes, is this supposed to be a secret then? If yes, how will we ensure some other app is not able to get this GUID.

Reply:
If the onus to keep the ID secret is on the app, would this ID based proposal be a relatively weaker construct? Someone can take a process dump of the user mode app and try to get the ID from there, and then use it in their app?

Reply:
How is this different from any other secret the app might need to maintain?